Parser/asdl_c.py

```diff
@@ -1132,13 +1132,291 @@ def visitModule(self, mod):
     return result;
 }
 
+/*
+ * Perform the following validations:
+ *
+ *   - All keyword arguments are known @e fields or @e attributes.
+ *   - No field or attribute would be left unfilled after copy.replace().
+ *
+ * On success, this returns 1. Otherwise, set a TypeError
+ * exception and returns -1 (no exception is set if some
+ * other internal errors occur).
+ *
+ * @param self          The AST node instance.
+ * @param dict          The AST node instance dictionary (self.__dict__).
+ * @param fields        The list of fields (self._fields).
+ * @param attributes    The list of attributes (self._attributes).
+ * @param kwargs        Keyword arguments passed to ast_type_replace().
+ *
+ * The @e dict, @e fields, @e attributes and @e kwargs arguments can be NULL.
+ */
+static inline int
+ast_type_replace_check(PyObject *self,
```

Comment:
Probably worth noting that this can be removed in 3.15, when the deprecations trigger and we can just rely on the constructor to do these checks.